spec/factories/stats.rb

```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :stat do
+    association :user, factory: :user, strategy: :build
+    type 'Stat::AnimeGenreBreakdown'
+    stats_data { { total: 0 } }
```

Comment:
First one is really more "ambiguous" than wrong, it could be either a block or a hash (and both are valid for FactoryGirl).

Also the problem is simple:

`{}.blank? #=> true`

You can probably drop the validation since the database provides a default